src/python/gudhi/differentiation/tensorflow.py

```diff
@@ -0,0 +1,258 @@
+                                                                 fcts, dtype=[tf.int32 for _ in range(2*c)]))
+        
+        # Get persistence diagram
+        self.dgm = tf.reshape(tf.gather_nd(self.F, inds), [c,2]) 
+        return self.dgm
+
+
+
+
+
+
+
+
+
+
+############################
+# Vietoris-Rips filtration #
+############################
+
+# The parameters of the model are the point coordinates.
+
+def _Rips(DX, mel, dim, card):
+    # Parameters: DX (distance matrix), 
+    #             mel (maximum edge length for Rips filtration), 
+    #             dim (homological dimension), 
+    #             card (number of persistence diagram points, sorted by distance-to-diagonal)
+
+    # Compute the persistence pairs with Gudhi
+    rc = RipsComplex(distance_matrix=DX, max_edge_length=mel)
+    st = rc.create_simplex_tree(max_dimension=dim+1)
+    dgm = st.persistence()
+    pairs = st.persistence_pairs()
+
+    # Retrieve vertices v_a and v_b by picking the ones achieving the maximal
+    # distance among all pairwise distances between the simplex vertices
+    indices, pers = [], []
+    for s1, s2 in pairs:
+        if len(s1) == dim+1 and len(s2) > 0:
+            l1, l2 = np.array(s1), np.array(s2)
+            i1 = [s1[v] for v in np.unravel_index(np.argmax(DX[l1,:][:,l1]),[len(s1), len(s1)])]
+            i2 = [s2[v] for v in np.unravel_index(np.argmax(DX[l2,:][:,l2]),[len(s2), len(s2)])]
+            indices += i1
+            indices += i2
+            pers.append(st.filtration(s2) - st.filtration(s1))
+    
+    # Sort points with distance-to-diagonal
+    perm = np.argsort(pers)
+    indices = list(np.reshape(indices, [-1,4])[perm][::-1,:].flatten())
+    
+    # Output indices
+    indices = indices[:4*card] + [0 for _ in range(0,max(0,4*card-len(indices)))]
+    return list(np.array(indices, dtype=np.int32))
+
+class RipsModel(tf.keras.Model):
+    """
+    TensorFlow model for computing Rips persistence out of a point cloud.
+
+    Attributes:
+        X (TensorFlow variable): point cloud of shape [number of points, number of dimensions]
+        mel (float): maximum edge length for the Rips complex 
+        card (int): maximum number of points in the persistence diagram
+        dim (int): homology dimension
+    """
+    def __init__(self, X, mel=12, dim=1, card=50):
+        super(RipsModel, self).__init__()
+        self.X = X
+        self.mel = mel
+        self.dim = dim
+        self.card = card
+        
+    def call(self):
+        m, d, c = self.mel, self.dim, self.card
+        
+        # Compute distance matrix
+        DX = tf.math.sqrt(tf.reduce_sum((tf.expand_dims(self.X, 1)-tf.expand_dims(self.X, 0))**2, 2))
+        DXX = tf.reshape(DX, [1, DX.shape[0], DX.shape[1]])
+        
+        # Turn numpy function into tensorflow function
+        RipsTF = lambda DX: tf.numpy_function(_Rips, [DX, m, d, c], [tf.int32 for _ in range(4*c)])
```

Comment:
Not sure if this is crucial, but `[tf.int32 for _ in range(4*c)]` can be replaced by `[tf.int32]*(4*c)`, which is about 30x faster with c=1000 for instance (same comment holds in other places as well).

Reply:
To reproduce my claim:
```python
~ ➜  ipython                                                                                                                                                                               
In [1]: import tensorflow as tf
In [2]: c = 1000
In [3]: %timeit [tf.int32 for _ in range(4*c)]
201 µs ± 5.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
In [4]: %timeit [tf.int32]*(4*c)
6.92 µs ± 16.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
In [5]: [tf.int32]*(4*c) == [tf.int32 for _ in range(4*c)]
Out[5]: True
```